File: apps/web/src/types/toolkit.ts
export type ToolkitMetrics = {
  sourceCount: number;
  toolCount: number;
  methodCount: number;
  intelligenceCount: number;
  evidenceCount: number;
  lastCollectedAt: string | null;
};

export type ToolkitTool = {
  id: string;
  name: string;
  category: string;
  riskLevel: string;
  collectorType: string;
  sourceTitle: string;
  sourceUrl: string | null;
  description: string | null;
  language: string | null;
  license: string | null;
  stars: number | null;
  forks: number | null;
  openIssues: number | null;
  updatedAt: string | null;
  collectedAt: string;
  sourceCredibilityScore: number;
  sourceCredibilityLevel: string;
  sourceCredibilityFactors: string[];
};

export type ToolkitMethod = {
  id: string;
  title: string;
  category: string;
  riskLevel: string;
  collectorType: string;
  sourceUrl: string | null;
  platform: string | null;
  recommendedCollector: string | null;
  dataTypes: string[];
  boundary: string | null;
  trainingTakeaway: string | null;
  collectedAt: string;
};

export type ToolkitIntelligence = {
  id: string;
  title: string;
  summary: string;
  domain: string;
  intelligenceType: string;
  finalScore: number;
  evidenceCount: number;
  updatedAt: string;
};

export type ToolkitLearningPath = {
  id: string;
  title: string;
  stage: string;
  focus: string;
  riskLevel: string;
  toolCount: number;
  methodCount: number;
  intelligenceCount: number;
  evidenceCount: number;
  tools: string[];
  methods: string[];
  acceptanceCriteria: string[];
  sourceUrls: string[];
};

export type ToolkitLecturePlaybook = {
  id: string;
  intelligenceId: string;
  title: string;
  audience: string;
  level: string;
  durationMinutes: number;
  claim: string;
  teachingSequence: string[];
  handsOnSteps: string[];
  verificationSteps: string[];
  riskBoundaries: string[];
  classroomExercise: string;
  evidenceUrls: string[];
  evidenceCount: number;
  finalScore: number;
};

export type ToolkitImageAnchorDiagnostic = {
  id: string;
  imageLabel: string;
  extractedClaim: string;
  sourceTitle: string;
  sourceUrl: string;
  sourceType: string;
  classification: string;
  riskLevel: string;
  valueJudgement: string;
  collectionUse: string;
  trainingTakeaway: string;
  relatedTools: string[];
  evidenceUrls: string[];
};

export type ToolkitBrowserLab = {
  id: string;
  title: string;
  focus: string;
  riskLevel: string;
  inspectionTargets: string[];
  playwrightChecks: string[];
  evidenceOutputs: string[];
  trainingTask: string;
  acceptanceCriteria: string[];
};

export type ToolkitAuthorizationChecklist = {
  id: string;
  title: string;
  riskLevel: string;
  requiredChecks: string[];
  blockedConditions: string[];
  evidenceRequired: string[];
  approvalRule: string;
};

export type ToolkitPreflightHttpResource = {
  url: string;
  statusCode: number | null;
  contentType: string | null;
  contentLength: number | null;
  available: boolean;
  summary: string;
};

export type ToolkitPreflightRedirect = {
  url: string;
  statusCode: number;
  location: string | null;
};

export type ToolkitPreflightDom = {
  title: string | null;
  description: string | null;
  canonicalUrl: string | null;
  metaRobots: string | null;
  headings: string[];
  linkCount: number;
  scriptCount: number;
  stylesheetCount: number;
  imageCount: number;
  formCount: number;
  textSample: string;
};

export type ToolkitPreflightNetwork = {
  requestMethod: string;
  finalStatusCode: number;
  finalContentType: string | null;
  redirectCount: number;
  sameOriginLinks: number;
  externalLinks: number;
  scriptCount: number;
  stylesheetCount: number;
  imageCount: number;
  formCount: number;
};

export type ToolkitPreflightAuthorizationGate = {
  allowedToContinue: boolean;
  riskLevel: string;
  blockedReasons: string[];
  requiredNextActions: string[];
};

export type ToolkitPreflightReport = {
  requestedUrl: string;
  finalUrl: string;
  checkedAt: string;
  authorizationConfirmed: boolean;
  headers: Record<string, string>;
  redirects: ToolkitPreflightRedirect[];
  robots: ToolkitPreflightHttpResource;
  sitemap: ToolkitPreflightHttpResource;
  securityTxt: ToolkitPreflightHttpResource;
  dom: ToolkitPreflightDom;
  network: ToolkitPreflightNetwork;
  authorizationGate: ToolkitPreflightAuthorizationGate;
  recommendations: string[];
};

export type ToolkitOverview = {
  dataset: string;
  generatedAt: string | null;
  metrics: ToolkitMetrics;
  learningPaths: ToolkitLearningPath[];
  lecturePlaybooks: ToolkitLecturePlaybook[];
  imageAnchorDiagnostics: ToolkitImageAnchorDiagnostic[];
  browserLabs: ToolkitBrowserLab[];
  authorizationChecklists: ToolkitAuthorizationChecklist[];
  tools: ToolkitTool[];
  methods: ToolkitMethod[];
  intelligenceItems: ToolkitIntelligence[];
};

File: apps/api/src/data_intelligence_hub/schemas/toolkit.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class ToolkitMetricsResponse(BaseModel):
    source_count: int
    tool_count: int
    method_count: int
    intelligence_count: int
    evidence_count: int
    last_collected_at: datetime | None


class ToolkitToolResponse(BaseModel):
    id: str
    name: str
    category: str
    risk_level: str
    collector_type: str
    source_title: str
    source_url: str | None
    description: str | None
    language: str | None
    license: str | None
    stars: int | None
    forks: int | None
    open_issues: int | None
    updated_at: datetime | None
    collected_at: datetime


class ToolkitMethodResponse(BaseModel):
    id: str
    title: str
    category: str
    risk_level: str
    collector_type: str
    source_url: str | None
    platform: str | None
    recommended_collector: str | None
    data_types: list[str]
    boundary: str | None
    training_takeaway: str | None
    collected_at: datetime


class ToolkitIntelligenceResponse(BaseModel):
    id: uuid.UUID
    title: str
    summary: str
    domain: str
    intelligence_type: str
    final_score: float
    evidence_count: int
    updated_at: datetime


class ToolkitLearningPathResponse(BaseModel):
    id: str
    title: str
    stage: str
    focus: str
    risk_level: str
    tool_count: int
    method_count: int
    intelligence_count: int
    evidence_count: int
    tools: list[str]
    methods: list[str]
    acceptance_criteria: list[str]
    source_urls: list[str]


class ToolkitLecturePlaybookResponse(BaseModel):
    id: str
    intelligence_id: uuid.UUID
    title: str
    audience: str
    level: str
    duration_minutes: int
    claim: str
    teaching_sequence: list[str]
    hands_on_steps: list[str]
    verification_steps: list[str]
    risk_boundaries: list[str]
    classroom_exercise: str
    evidence_urls: list[str]
    evidence_count: int
    final_score: float


class ToolkitOverviewResponse(BaseModel):
    dataset: str
    generated_at: datetime | None
    metrics: ToolkitMetricsResponse
    learning_paths: list[ToolkitLearningPathResponse]
    lecture_playbooks: list[ToolkitLecturePlaybookResponse]
    tools: list[ToolkitToolResponse]
    methods: list[ToolkitMethodResponse]
    intelligence_items: list[ToolkitIntelligenceResponse]
